base/compiler/tfuncs.jl

```diff
@@ -925,11 +925,11 @@ function fieldtype_tfunc(@nospecialize(s0), @nospecialize(name))
     if s0 === Any || s0 === Type || DataType ⊑ s0 || UnionAll ⊑ s0
         return Type
     end
-    # fieldtype only accepts Types, errors on `Module`
-    if isa(s0, Const) && (!(isa(s0.val, DataType) || isa(s0.val, UnionAll) || isa(s0.val, Union)) || s0.val === Module)
+    # fieldtype only accepts Types
+    if isa(s0, Const) && !(isa(s0.val, DataType) || isa(s0.val, UnionAll) || isa(s0.val, Union))
         return Bottom
     end
-    if s0 == Type{Module} || s0 == Type{Union{}} || isa(s0, Conditional)
+    if (s0 isa Type && s0 == Type{Union{}}) || isa(s0, Conditional)
```

Comment:
@Sacha0 - i also did this backport, here (oops, we collided. I'm happy to drop that now that you've done it 😊):
https://github.com/JuliaLang/julia/compare/backports-release-1.5...NHDaly:nhd-backports-release-1.5-manual

And i wasn't sure whether to pull over the `isa Type` change as well. Technically that was added in https://github.com/JuliaLang/julia/commit/dcc0696971a100dbc63f90b8789632631dd1bfef, as part of (#36596), so i wasn't sure whether to backport it as well. Supposedly it's just a code cleanup, but i'm not sure if it holds in isolation from the rest of the commit.

I think the closest transformation keeping in spirit of the change we're backporting from (#37645) would be just to drop the `Module` part, since that's all they did in the original PR? Like this, perhaps?:
```suggestion
    if s0 == Type{Union{}} || isa(s0, Conditional)
```

PROBABLY fine either way, but since i see you made a different decision than I made when attempting the manual bisect, I wanted to just bring this up! :)

Reply:
To see the diff standalone, it's the change i made:
https://github.com/JuliaLang/julia/compare/backports-release-1.5...NHDaly:nhd-backports-release-1.5-manual#diff-94ffb7fc1a419bb8439ae0ecbb4a3b5460d193e53c96bb97b50e283fdbc16a85L932-R932

Reply:
EDIT: https://github.com/JuliaLang/julia/commit/cc099114749a53490d7903dc122dd813609ea569#diff-94ffb7fc1a419bb8439ae0ecbb4a3b5460d193e53c96bb97b50e283fdbc16a85L932-R932

Reply:
I wondered the same, but err'd to adding the `isa Type` guard as `== Type{Union{}}` cannot be true unless the guard is true, and if the latter is true then the former must be true as well, so this transformation should be functionally identical modulo some additional safety with the guard (against `== Type{Union{}}` throwing something for non-`Type` `s0`s)? Either way is fine by me though :).